x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -789,7 +789,22 @@ private PhysicalOperation planLookupJoin(LookupJoinExec join, LocalExecutionPlan
             if (input == null) {
                 throw new IllegalArgumentException("can't plan [" + join + "][" + left + "]");
             }
-            matchFields.add(new MatchConfig(right, input));
+            // we support 2 types of joins: Field name joins and Expression joins
+            // for Field name join, we do not ship any join on expression.
+            // we built the Lucene query on the field name that is passed in the MatchConfig.fieldName
+            // so for Field name we need to pass the attribute name from the right side, because that is needed to build the query
+            // For expression joins, we pass an expression such as left_id > right_id.
+            // So in this case we pass in left_id as the field name, because that is what we are shipping to the lookup node
+            // The lookup node will replace that name, with the actual values for each row and perform the lookup join
+            // We need to pass the left name, because we need to know what data we have shipped.
+            // It is not acceptable to just use the left or right side of the operator because the same field can be joined multiple times
+            // e.g. LOOKUP JOIN ON left_id < right_id_1 and left_id >= right_id_2
+            // we want to be able to optimize this in the future and only ship the left_id once
+            FieldAttribute.FieldName fieldName = right.exactAttribute().fieldName();
```

Comment:
I'd leave the comment that was on the original code: the call to `exactAttribute()` is likely redundant because we don't (yet) work with TEXT fields in the lookup index.